### PR TITLE
Use fetch-mock instead of nock

### PR DESCRIPTION
**Pull request by ranquild**
Related to https://github.com/metabase/metabase/issues/28489

Problems with `nock`:
- It doesn't play well with `jest`. There are known memory and performance issues that are not solved for a year https://github.com/nock/nock/issues/2358, https://github.com/nock/nock/issues/2383
- It turned out to be heavyweight. It patches node `http` and `https` module on a very low level. What we actually need is something that overwrites the global `fetch` function with something mock-friendly.

[fetch-mock](https://www.wheresrhys.co.uk/fetch-mock/) is a popular library for request mocking, but it works only with `fetch`. And if you try to use `fetch` with our cypress tests, about ~20-40 of them fail for no obvious reason, which makes me think that cypress handles xhr and fetch differently. So the solution I came up with is to use `fetch` in tests only for now and migrate from `nock` asap because it becomes a real CI blocker.

This PR converts just one single test as a proof of concept. We can migrate our existing `nock`-based tests in follow-up PRs.

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/28481)

### CODE REVIEW

File: frontend/test/metabase/lib/api.unit.spec.js [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
   afterEach(() => {
     nock.cleanAll();
+    fetchMock.reset();
   });
 
   const successResponse = { status: "ok" };
 
   it("should GET", async () => {
-    nock(location.origin).get("/hello").reply(200, successResponse);
+    fetchMock.get("path:/hello", { status: 200, body: successResponse });
```
Comment (ranquild) [inline frontend/test/metabase/lib/api.unit.spec.js:15]: `:path` matcher allows us to skip the origin, which is an improvement over `nock` already :)

**Review by alxnddr — APPROVED** [review]
I like the plan! Later we can add https://www.npmjs.com/package/fetch-mock-jest

File: frontend/src/metabase/lib/api.js [hunk 2]
```diff
@@ -152,8 +153,15 @@ export class Api extends EventEmitter {
...
 
-  // TODO Atte Keinänen 6/26/17: Replacing this with isomorphic-fetch could simplify the implementation
-  _makeRequest(method, url, headers, body, data, options) {
+  _makeRequest(...args) {
+    if (isTest) {
+      return this._makeRequestWithFetch(...args);
+    } else {
+      return this._makeRequestWithXhr(...args);
+    }
+  }
```
Comment (kulyk) [inline frontend/src/metabase/lib/api.js:162]: I think it's worth a comment; we can link this PR here

**Review by kulyk — APPROVED** [review]
LGTM 👍

**Comment by WiNloSt** [comment]
One thing that might help you track down future work when we're migrating everything to fetch is, I randomly picked a few of the failed tests [from this comment](https://github.com/metabase/metabase/pull/28481#issuecomment-1438488158) (for future readers you might need to look at the comment history, currently it's 19:40 GMT)

From this screenshot
image

You can see that Cypress doesn't type the first few characters (I found [another test](https://www.deploysentinel.com/ci/runs/63f62e8315d18167a438a716) that missed the first 2 characters). IIRC, I've encountered this problem before in my test a while ago, the cause at that time was like something triggered the rerender and causes the first few keystrokes to disappear.

I'm not sure what it is this time, but the symptom is very similar.

**Pull request merged**